spark/src/main/scala/io/ddf/spark/analytics/AggregationHandler.scala

```diff
@@ -0,0 +1,40 @@
+package io.ddf.spark.analytics
+
+import io.ddf.DDF
+import io.ddf.analytics.{AggregationHandler=>CoreAggregationHandler}
+import io.ddf.exception.DDFException
+import java.util.{List=>JList}
+
+import io.ddf.spark.util.SparkUtils
+import io.ddf.types.AggregateTypes
+import io.ddf.types.AggregateTypes.{AggregateField, AggregationResult}
+import org.apache.spark.sql.AnalysisException
+
+import scala.collection.JavaConverters._
+
+/**
+ * Created by nhanitvn on 3/21/16.
+ */
+class AggregationHandler(theDDF: DDF) extends CoreAggregationHandler(theDDF) {
+  override def groupBy(groupedColumns: JList[String], aggregateFunctions: JList[String]): DDF = {
+    // Catch SparkSQL/JSQLParser error message, parse and produce appropriate error message
+    try {
+      super.groupBy(groupedColumns, aggregateFunctions)
+    } catch {
+      case ddfException: DDFException => throw new DDFException(SparkUtils.sqlErrorToDDFError(ddfException.getMessage, buildGroupBySQL(aggregateFunctions), aggregateFunctions.asScala.toList:::groupedColumns.asScala.toList))
+      case throwable: Throwable => throw throwable
```

Comment:
This case clause is unnecessary. You don't have to catch if you just throw it again as that's the default behaviours.